### PR TITLE
Run parallel unit tests when possible and update codecov

### PR DESCRIPTION
**Pull request by BrandonWeng**

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
       - "**.go"
     branches:
       - main
+      - master
```
Comment (BrandonWeng) [inline .github/workflows/unit_tests.yml:9]: We haven't been running the unit tests on master branch

File: Makefile [hunk 2]
```diff
@@ -197,4 +197,4 @@ $(BUILDDIR)/packages.txt:$(GO_TEST_FILES) $(BUILDDIR)
 split-test-packages:$(BUILDDIR)/packages.txt
 	split -d -n l/$(NUM_SPLIT) $< $<.
 test-group-%:split-test-packages
-	cat $(BUILDDIR)/packages.txt.$* | xargs go test -mod=readonly -timeout=10m -race -coverprofile=$(BUILDDIR)/$*.profile.out
+	cat $(BUILDDIR)/packages.txt.$* | xargs go test -parallel 4 -mod=readonly -timeout=10m -race -coverprofile=$(BUILDDIR)/$*.profile.out
```
Comment (BrandonWeng) [inline Makefile:200]: Some unit tests have parallel enabled, this is just so that they can run in parallel

Comment (BrandonWeng) [reply Makefile:200]: Technically should speed up our test runtime too

File: .github/workflows/unit-test-coverage.yml [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
-
+name: Unit Test Coverage
 
 on:
   pull_request:
-    branches: [ master ]
+  push:
+    branches:
+      - master
+      - release/**
```
Comment (BrandonWeng) [inline .github/workflows/unit-test-coverage.yml:8]: I think the issue is that we're not running on the master branch (base commit that PRs are forked from)

image

**Review by philipsu522 — APPROVED** [review]

**Review by yzang2019 — APPROVED** [review]

**Pull request merged**